pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -63,9 +63,14 @@ func New(mounterPath string) *Mounter {
 
 func (m *Mounter) Mount(ctx context.Context, mc *MountConfig) error {
 	// Start the token server for HostNetwork enabled pods.
-	if mc.TokenServerIdentityProvider != "" {
-		klog.Infof("Pod has hostNetwork enabled and token server feature is turned on. Starting Token Server on %s/%s", mc.TempDir, TokenFileName)
-		go StartTokenServer(ctx, mc.TempDir, TokenFileName, mc.TokenServerIdentityProvider)
+	// For managed sidecar, the token server identity provider is only populated when host network pod ksa feature is opted in.
```

Comment:
this comment can be removed. since now a user can also provide an identity provider with a managed sidecar version also, 

```
shouldPopulateIdentifyProvider() {
   ...
   tokenVolumeInjected && (sidecarVersionSupported || userInput) // returns true
}
```
```
	if s.shouldPopulateIdentifyProvider(pod, optInHostnetworkKSA, userSpecifiedIdentityProvider != "") {
		identityProvider := ""
siyanshen marked this conversation as resolved.
		if userSpecifiedIdentityProvider != "" {
			identityProvider = userSpecifiedIdentityProvider
		} else {
			identityProvider = s.driver.config.TokenManager.GetIdentityProvider()
		}
		fuseMountOptions = joinMountOptions(fuseMountOptions, []string{"hnw-ksa=true", "token-server-identity-provider=" + identityProvider})
	}
```